src/lib/chains/useRealChainIdWarning.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useSyncExternalStore } from "react";
+import { toast } from "react-toastify";
+import { useAccount } from "wagmi";
+
+import { useChainId as useAssumedChainId } from "lib/chains";
+import { INVALID_NETWORK_TOAST_ID, getInvalidNetworkErrorMessage } from "lib/contracts/transactionErrors";
+
+const toastSubscribe = (onStoreChange: () => void): (() => void) => {
+  const cleanup = toast.onChange(({ id }) => {
+    if (id === INVALID_NETWORK_TOAST_ID) {
+      onStoreChange();
+    }
+  });
+
+  return cleanup;
+};
+
+const toastGetSnapshot = () => toast.isActive(INVALID_NETWORK_TOAST_ID);
+
+export function useRealChainIdWarning() {
+  const { isConnected, chainId } = useAccount();
+  const { chainId: assumedChainId } = useAssumedChainId();
+
+  const isActive = useSyncExternalStore(toastSubscribe, toastGetSnapshot);
+
+  useEffect(() => {
+    if (
+      chainId !== undefined &&
+      assumedChainId !== undefined &&
+      chainId !== assumedChainId &&
```

Comment:
my proposal:
```
useAssumedChainId -> useChainId
fakeChainId -> displayedChainId
```
useChainId also returns `isSupportedChainId` which equals to 
```
chainId !== undefined &&
displayedChainId !== undefined &&
chainId !== displayedChainId 
```

`isSupportedChainId` is used in this effect

Reply:
Agree, should be prettier